pkg/virtctl/ssh/knownhosts_test.go

```diff
@@ -0,0 +1,52 @@
+package ssh
+
+import (
+	"bufio"
+	"crypto/rand"
+	"crypto/rsa"
+	"os"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"golang.org/x/crypto/ssh"
+)
+
+var _ = Describe("Known Hosts", func() {
+
+	var tmpDir string
+	BeforeEach(func() {
+		tmpDir = GinkgoT().TempDir()
+	})
+
+	It("should be added with a newline", func() {
+		knowHostFile := filepath.Join(tmpDir, "knownhosts")
+		f, err := os.Create(knowHostFile)
+		Expect(err).ToNot(HaveOccurred())
+		f.Close()
+
+		privateKey, err := rsa.GenerateKey(rand.Reader, 1024)
+		Expect(err).ToNot(HaveOccurred())
+		// generate and write public key
+		pub, err := ssh.NewPublicKey(&privateKey.PublicKey)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(addHostKey(knowHostFile, "host1", pub)).To(Succeed())
+		Expect(addHostKey(knowHostFile, "host2", pub)).To(Succeed())
+		Expect(addHostKey(knowHostFile, "host3", pub)).To(Succeed())
+
+		f, err = os.Open(knowHostFile)
+		Expect(err).ToNot(HaveOccurred())
+		scanner := bufio.NewScanner(f)
+
+		lineCount := 0
+		for {
+			if !scanner.Scan() {
+				break
+			}
+			lineCount++
+		}
```

Comment:
worth to extract to a function please

Reply:
I agree; while  not a must, I think this would make the test easier to read:

```golang
func numberOfLinesInFile(filePath string) (int, error) {
...
}

# caller
Expect(numberOfLinesInFile(knowHostFile)).Should(Equal(3))
```

Reply:
Done.